src/context/sessionContext/index.test.tsx

```diff
@@ -262,4 +263,31 @@ describe("SessionContext functionality", () => {
     fireEvent.click(getByText("Logout"));
     await waitFor(() => expect(onError).toHaveBeenCalledTimes(1));
   });
+
+  it("registers a session restore callback if one is provided", async () => {
+    (handleIncomingRedirect as jest.Mock).mockResolvedValueOnce(null);
+    (onSessionRestore as jest.Mock).mockResolvedValueOnce(null);
+
+    const session = {
+      info: {
+        isLoggedIn: true,
+        webId: "https://fakeurl.com/me",
+      },
+      on: jest.fn(),
+    } as any;
+
+    (getDefaultSession as jest.Mock).mockReturnValue(session);
+
+    render(
+      <SessionProvider sessionId="key" onSessionRestoreCallback={() => {}}>
+        <ChildComponent />
+      </SessionProvider>
+    );
+
+    await waitFor(() => {
+      expect(handleIncomingRedirect).toHaveBeenCalled();
+    });
+
+    expect(onSessionRestore).toHaveBeenCalled();
```

Comment:
For extra thoroughness, a check that it actually registers the passed callback:

```suggestion
    const sessionRestoreCallback = jest.fn();
    render(
      <SessionProvider sessionId="key" onSessionRestoreCallback={sessionRestoreCallback}>
        <ChildComponent />
      </SessionProvider>
    );

    await waitFor(() => {
      expect(handleIncomingRedirect).toHaveBeenCalled();
    });

    expect(onSessionRestore).toHaveBeenCalledWith(sessionRestoreCallback);
```